src/server.rs

```diff
@@ -1268,9 +1270,10 @@ impl proto::Peer for Peer {
 
         let mut request = match b.body(()) {
             Ok(request) => request,
-            Err(_) => {
+            Err(e) => {
                 // TODO: Should there be more specialized handling for different
                 // kinds of errors
+                trace!("request body error: {}, PROTOCOL_ERROR", e);
```

Comment:
note that this isn't necessarily a _body errror_ -- it's any error building the request.

Reply:
ah, you're right, will rephrase.